### PR TITLE
Requirements by Jenkins marketplace

### PR DESCRIPTION
**Pull request by BirSikanderMahajan**
Updating the repo as per requirements by Jenkins marketplace, 
- have included spotbugs fixation.
- Added SNAPSHOT version into repo

### CODE REVIEW

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 1]
```diff
@@ -468,7 +476,12 @@ public void createJUnitReport(FilePath workspace, String JUnitReportFile, final
             {
                 File file = new File(workspace.toURI().getPath(),JUnitReportFile);
                 listener.getLogger().println(String.format(Messages.FULL_REPORT_FILE_PATH,file.getCanonicalPath()));
-                if(!file.exists()) file.createNewFile();
+              if(!file.exists()) {
+            	  try {file.createNewFile();}
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:480]: this section is an exampe of really bad styling
```java
  try {
    file.createNewFile();
  }
  catch (Exception e) {
    throw e;
  }
}
```

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 2]
```diff
@@ -34,9 +36,9 @@
 
     private PluginHandler(){}
 
-    public static PluginHandler getInstance()
+    public synchronized static PluginHandler getInstance()
     {
-        if( pluginHandler == null ) pluginHandler = new PluginHandler();
+        if( pluginHandler == null ) { pluginHandler = new PluginHandler();}
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:41]:
```java
if (pluginHandler == null)
{
  pluginHandler = new PluginHandler();
}
```

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 3]
```diff
@@ -237,7 +239,7 @@ public String getWorkSpaceSafe(FilePath workspace, EnvVars env)
                     StringBuilder errorMessage401 = new StringBuilder(String.format(Messages.ERROR_CODE_MESSAGE, response.getStatusCode(), response.getStatusText()));
                     appendLine(errorMessage401,Messages.INVALID_ACCESS_KEY);
                     OnFailedToGetScheduleTitleIdMap(null,errorMessage401.toString(),listener);
-
+break;
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:242]: indent this line

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 4]
```diff
@@ -246,7 +248,7 @@ public String getWorkSpaceSafe(FilePath workspace, EnvVars env)
                 default:
                     StringBuilder errorMessage = new StringBuilder(String.format(Messages.ERROR_CODE_MESSAGE, response.getStatusCode(), response.getStatusText()));
                     OnFailedToGetScheduleTitleIdMap(null,errorMessage.toString(),listener);
-
+                    throw new Exception(errorMessage.toString());
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:251]: `OnFailedToGetScheduleTitleIdMap` is throwing an exception already, check the code

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 5]
```diff
@@ -442,10 +444,16 @@ public boolean stopRun(String controllerApiHttpAddress, UUID runId, String sched
                     listener.error(String.format(Messages.STOP_RUN_FAIL,scheduleTitle,runId.toString()));
 
             }
-        } catch (Exception e)
+        } 
+        
+        catch (RuntimeException e) {
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:449]: what is the reason to catch `RuntimeException` to throw it again wrapped into another `RuntimeException`?

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 6]
```diff
@@ -442,10 +444,16 @@ public boolean stopRun(String controllerApiHttpAddress, UUID runId, String sched
...
+			throw new RuntimeException(e);
+		}
+        catch (Exception e)
         {
             listener.error(String.format(Messages.STOP_RUN_FAIL,scheduleTitle,runId.toString()));
             listener.error(e.getMessage());
+           throw e;
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:456]: throwing this error leads to losing some logic in the calling method, check `CollectScheduleRunResults`

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 7]
```diff
@@ -658,7 +672,7 @@ public RunItem getRunItem(AsyncHttpClient client, String controllerApiHttpAddres
                 else
                 {
                     Failure keyframes = getRunItemKeyFrames(client,controllerApiHttpAddress,accessKey,runItemId,runItem,scheduleTitle,environmentTitle,listener);
-                    runItem.failure = keyframes;
+					/* runItem.failure = keyframes; */
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:675]: so failed information is not needed anymore?

**Review by AntonLiakhovich — CHANGES_REQUESTED** [review]
Check all comments

File: src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java [hunk 8]
```diff
@@ -28,15 +28,15 @@ public RunItem(String caseTitle, String caseStatus, double elapsed, String sched
         this.caseStatus = caseStatus;
         elapsedTime = elapsed;
         classname = schedule;
-        failure = null;
```
Comment (RomanSyrychLeapwork) [inline src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java:31]: Without it JUnit report in post build actions will not show keyframes. Why is it removed?

File: src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java [hunk 9]
```diff
@@ -28,15 +28,15 @@ public RunItem(String caseTitle, String caseStatus, double elapsed, String sched
...
+
     }
 
     public RunItem(String caseTitle, String caseStatus, double elapsed, String stacktrace, String schedule)
     {
         caseName = caseTitle;
         this.caseStatus = caseStatus;
         elapsedTime = elapsed;
-        failure = new Failure(stacktrace);
```
Comment (RomanSyrychLeapwork) [inline src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java:39]: The same: Without it JUnit report in post build actions will not show keyframes. Why is it removed?

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 7 above)
Comment (RomanSyrychLeapwork) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:675]: Yeah, all keyframe messages are missing. That's one of the most important places in the plugin

File: src/main/java/com/Leapwork/Leapwork_plugin/LeapworkJenkinsBridgeBuilder.java [hunk 10]
```diff
@@ -327,6 +327,7 @@ private static void CollectScheduleRunResults(String controllerApiHttpAddress, S
 							resultRun.incFailed();
 						resultRun.incTotal();
 						break;
+						default:
```
Comment (RomanSyrychLeapwork) [inline src/main/java/com/Leapwork/Leapwork_plugin/LeapworkJenkinsBridgeBuilder.java:330]: I would rather marked a run item with undefined/unexpected status in this case

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 11]
```diff
@@ -602,7 +616,7 @@ public String getRunStatus(AsyncHttpClient client, String controllerApiHttpAddre
 
         }
     }
-
+    @SuppressFBWarnings(value = "DLS_DEAD_LOCAL_STORE")
```
Comment (RomanSyrychLeapwork) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:619]: What is it and why it is needed?

**Review by RomanSyrychLeapwork — CHANGES_REQUESTED** [review]
Please resolve existing issues. At current moment we can't pass it to main branch

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 3 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:242]: resolved

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 4 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:251]: removed

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 5 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:449]: removing it

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 11 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:619]: this was required for suppressing given by spotbugs in maven,

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java (hunk 7 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:675]: removed comments, added code into repo

File: src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java (hunk 9 above)
Comment (BirSikanderMahajan) [reply src/main/java/com/Leapwork/Leapwork_plugin/model/RunItem.java:39]: added again

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 12]
```diff
@@ -23,781 +25,759 @@
...
+				appendLine(errorMessage401, Messages.INVALID_ACCESS_KEY);
+				OnFailedToGetScheduleTitleIdMap(null, errorMessage401.toString(), listener);
+				break;
+			case 500:
+				StringBuilder errorMessage500 = new StringBuilder(
+						String.format(Messages.ERROR_CODE_MESSAGE, response.getStatusCode(), response.getStatusText()));
+				appendLine(errorMessage500, Messages.CONTROLLER_RESPONDED_WITH_ERRORS);
+				OnFailedToGetScheduleTitleIdMap(null, errorMessage500.toString(), listener);
+
+			default:
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:224]: don't you need `break;` before `default` as a final step for `case 500:`?

File: src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java [hunk 13]
```diff
@@ -23,781 +25,759 @@
...
+				JsonObject jsonStopRunObject = parser.parse(response.getResponseBody()).getAsJsonObject();
+				JsonElement jsonStopSuccessfull = jsonStopRunObject.get("OperationCompleted");
+				isSuccessfullyStopped = Utils.defaultBooleanIfNull(jsonStopSuccessfull, false);
+				if (isSuccessfullyStopped)
+					listener.error(String.format(Messages.STOP_RUN_SUCCESS, scheduleTitle, runId.toString()));
+				else
+					listener.error(String.format(Messages.STOP_RUN_FAIL, scheduleTitle, runId.toString()));
+				break;
+
+			case 401:
```
Comment (AntonLiakhovich) [inline src/main/java/com/Leapwork/Leapwork_plugin/PluginHandler.java:376]: `break` after each `case`?

**Review by AntonLiakhovich — CHANGES_REQUESTED** [review]

**Review by AntonLiakhovich — APPROVED** [review]

**Review by RomanSyrychLeapwork — APPROVED** [review]

**Pull request merged**